src/stores/search.js

```diff
@@ -0,0 +1,172 @@
+ * `categories` can mean `imageCategories` if the current `searchType` is `IMAGE`, or `audioCategories`
+ * if it is `AUDIO`.
+ * - query parameter values are the values that are used in the API query.
+ */
+export const useSearchStore = defineStore('search', () => {
+  const filterStore = useFilterStore()
+  /**
+   * @type {import('../store/types').SearchState}}
+   */
+  const state = reactive({
+    searchType: ALL_MEDIA,
+    query: {
+      q: '',
+      license_type: '',
+      license: '',
+      categories: '',
+      extension: '',
+      duration: '',
+      aspect_ratio: '',
+      size: '',
+      source: '',
+      searchBy: '',
+      // mature can be either 'true' or ''
+      mature: '',
+    },
+  })
+  const { searchType, query } = toRefs(state)
+
+  /**
+   * @param {import('../store/types').SupportedSearchType} type
+   */
+  const setSearchType = (type) => {
+    state.searchType = type
+    filterStore.setSearchType(type)
+  }
+
+  /**
+   * Returns the search query parameters for API request:
+   * drops all parameters with blank values.
+   *
+   * @returns {import('@nuxtjs/composition-api').ComputedRef<import('../store/types').ApiQueryParams>}
+   */
+  const searchQueryParams = computed(() => {
+    // Ensure that q filter always comes first
+    return Object.keys(state.query).reduce(
+      (obj, key) => {
+        return !['q'].includes(key) && state.query[key].length
+          ? { ...obj, [key]: state.query[key] }
+          : obj
+      },
+      { q: state.query.q.trim() }
+    )
+  })
+
+  /**
+   * Called when `q` search term or `searchType` are changed. Updates the filters for
+   * the current search type, and then updates the search query accordingly.
+   * @param {{ q?: string, searchType?: import('../store/types').SupportedSearchType }} params
+   */
+  function updateQuery(params = {}) {
+    const { q, searchType } = params
+    if (q) {
+      state.query.q = q.trim()
+    }
+    if (searchType && searchType !== state.searchType) {
+      setSearchType(searchType)
+      filterStore.clearOtherMediaTypeFilters({ searchType })
+    }
+    updateQueryFromFilters()
+  }
+
+  /**
+   * Called when a /search path is server-rendered.
+   * @param {{ path: string, query: { [key: string]: string} }} params
+   */
+  function setSearchStateFromUrl({ path, query }) {
+    if (query.q) {
+      state.query.q = query.q.trim()
+    }
+
+    const searchType = queryStringToSearchType(path)
+    setSearchType(searchType)
+
+    filterStore.updateFiltersFromUrl(query, searchType)
+    updateQueryFromFilters()
+  }
+  /**
+   * After a change in filters, updates the query. If the filter is not selected, query should be set to blank string "".
+   * @param {object} [params]
+   * @param {import('../store/types').SupportedSearchType} [params.mediaType]
+   * @param {string} [params.q]
+   */
+  function updateQueryFromFilters(params = {}) {
```

Comment:
It does not appear that an argument is ever passed to this function when it is called. Can we remove the params if that's the case?

```suggestion
   */
  function updateQueryFromFilters() {
```